source/extensions/filters/http/jwt_authn/header_matchers.h

```diff
@@ -19,25 +19,34 @@ namespace JwtAuthn {
  */
 class HeaderMatcher : public Matcher, public Logger::Loggable<Logger::Id::filter> {
 public:
-  HeaderMatcher(const ::envoy::api::v2::route::RouteMatch& route);
+  HeaderMatcher(const ::envoy::config::filter::http::jwt_authn::v2alpha::RequirementRule& rule,
```

Comment:
Again,  Matcher is an interface.  all others are its implemenation.  so call them 

xxxMatcherImpl